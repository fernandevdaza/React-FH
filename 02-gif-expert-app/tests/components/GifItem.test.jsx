const { render, screen } = require('@testing-library/react');
const { GifItem } = require('../../src/components');

describe('Tests on <GifItem />', () => {
    const title = 'Hello';
    const url = 'https://www.google.com/';
    test('should match with snapshot', () => {
        const { container } = render(<GifItem title={ title } url={ url } />);
        expect(container).toMatchSnapshot();
    });
    test('should show image with the exact url and alt', () => {
        render(<GifItem title={ title } url={ url } />);
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });
    test('should show title in component', () => {
        render(<GifItem title={ title } url={ url } />);
        expect(screen.getByText(title)).toBeTruthy();
    });
});
