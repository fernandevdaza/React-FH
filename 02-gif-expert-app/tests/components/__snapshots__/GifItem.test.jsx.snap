// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifItem /> should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Hello"
      src="https://www.google.com/"
    />
    <p>
      Hello
    </p>
  </div>
</div>
`;
